piet-scene/src/scene/builder.rs

```diff
@@ -134,30 +131,35 @@ impl<'a> SceneBuilder<'a> {
 
     /// Completes construction and finalizes the underlying scene.
     pub fn finish(mut self) {
-        while let Some(layer) = self.layers.pop() {
-            self.end_clip(Some(layer));
+        while !self.layers.is_empty() {
+            self.pop_layer();
         }
     }
 }
 
 impl<'a> SceneBuilder<'a> {
-    fn encode_path<E>(&mut self, elements: E, is_fill: bool) -> bool
-    where
-        E: Iterator,
-        E::Item: Borrow<PathElement>,
-    {
+    fn encode_path(&mut self, shape: &impl Shape, is_fill: bool) -> bool {
```

Comment:
This is another place where I'd like a comment explaining the bool result.

Reply:
A further thought on this: when reading through kurbo code, I noticed that it treats a line/quad/curve at the start of a path as a move. This code currently assumes an implicit move to (0,0) in that case. Should I update this to be consistent with kurbo?

Reply:
I'm slightly confused what you're asking. In `BezPath` builder methods, there's a debug assert against that (I consider it an invalid state). I think a reason to have an implicit move of (0, 0) is so you don't get an invalidly encoded path as output even when the input is wrong, but the exact value of what you get is not too important, people shouldn't be relying on it.

But perhaps I'm just misunderstanding.

Reply:
It specifically noted for the Shape impl for slices here: https://github.com/linebender/kurbo/blob/fd839c25ea0c98576c7ce5789305822675a89938/src/bezpath.rs#L1177

The logic is implemented in the Segments iterator. FWIW, I think SVG does the same but I’d have to check the spec.

Reply:
Ah yes, this behavior was changed in https://github.com/linebender/kurbo/pull/98. That particular use case doesn't seem too important here, but probably having fewer different behaviors is better.